src/VisualStudio/Core/Def/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -155,30 +155,19 @@ public async Task WaitUntilFullyLoadedAsync(CancellationToken cancellationToken)
                 }
             }
 
-            // unfortunately, IVsOperationProgressStatusService requires UI thread to let project system to proceed to next stages.
-            // this method should only be used with either await or JTF.Run, it should be never used with Task.Wait otherwise, it can
-            // deadlock
             public async Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
             {
                 var status = await GetProgressStageStatusAsync(cancellationToken).ConfigureAwait(false);
-                if (status == null)
-                {
-                    return false;
-                }
-
-                return !status.IsInProgress;
+                return status != null && !status.IsInProgress;
             }
 
             private async ValueTask<IVsOperationProgressStageStatusForSolutionLoad?> GetProgressStageStatusAsync(CancellationToken cancellationToken)
             {
                 // Workaround for lack of fast path in JoinAsync; avoid calling when already completed
                 // https://github.com/microsoft/vs-threading/pull/696
```

Comment:
microsoft/vs-threading#696 was merged, so this no longer needs a workaround. Just use `JoinAsync` on both paths.

Reply:
@sharwell  do i need 'paths' at all at that point?  should just just be `await _progressStageStatus.JoinAsync(cancellationToken).ConfigureAwait(false)`?

Reply:
Right that's what I meant 😄

Reply:
perfect.  thanks.